### PR TITLE
DTSPO-0381 Using Tags module

### PR DESCRIPTION
**Pull request by mokainos**
https://tools.hmcts.net/jira/browse/DTSPO-0381

### CODE REVIEW

File: components/backendappgateway/local.tf [hunk 1]
```diff
@@ -1,41 +1,23 @@
...
-    prod     = "Production"
-    demo     = "Demo"
-    ithc     = "ITHC"
-    ldata    = "LDATA"
-    perftest = "Test"
-    preview  = "Dev"
-    dev      = "Dev"
-    test     = "Test"
-    stg      = "Staging"
+  log_analytics_env_mapping = {
```
Comment (timja) [inline components/backendappgateway/local.tf:3]: why is this added here? it wasn't here before

File: pipeline-templates/terraform.yaml [hunk 2]
```diff
@@ -40,7 +40,7 @@ steps:
       terraformVersion: ${{ parameters.tfversion }}
       workingDirectory: $(System.DefaultWorkingDirectory)/components/${{ parameters.stack }}
       environmentServiceName: ${{ parameters.serviceConnection }}
-      commandOptions: '-out tfplan-${{ parameters.project }}-${{ parameters.environment }}-${{ parameters.stack }}  -var environment=${{ parameters.environment }} -var project=${{ parameters.project }} -var-file $(System.DefaultWorkingDirectory)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars -lock=false'
+      commandOptions: '-out tfplan-${{ parameters.project }}-${{ parameters.environment }}-${{ parameters.stack }}  -var environment=${{ parameters.environment }} -var project=${{ parameters.project }} -var builtFrom=$(Build.Repository.Name) -var-file $(System.DefaultWorkingDirectory)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars -lock=false'
```
Comment (timja) [inline pipeline-templates/terraform.yaml:43]: missing var.product

File: pipeline-templates/terraform.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ steps:
       terraformVersion: ${{ parameters.tfversion }}
       workingDirectory: $(System.DefaultWorkingDirectory)/components/${{ parameters.stack }}
       environmentServiceName: ${{ parameters.serviceConnection }}
-      commandOptions: '-out tfplan-${{ parameters.project }}-${{ parameters.environment }}-${{ parameters.stack }}  -var environment=${{ parameters.environment }} -var project=${{ parameters.project }} -var builtFrom=$(Build.Repository.Name) -var-file $(System.DefaultWorkingDirectory)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars -lock=false'
+      commandOptions: '-out tfplan-${{ parameters.project }}-${{ parameters.environment }}-${{ parameters.stack }}  -var environment=${{ parameters.environment }} -var project=${{ parameters.project }} -var builtFrom=$(Build.Repository.Name) -var product=sds-platform -var-file $(System.DefaultWorkingDirectory)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars -lock=false'
```
Comment (timja) [inline pipeline-templates/terraform.yaml:43]: product should be passed in from higher up and only interpreted as a variable here, too easy to copy paste this and keep the parameter the same when it shouldn't be

File: pipeline-templates/terraform.yaml (hunk 2 above)
Comment (mokainos) [reply pipeline-templates/terraform.yaml]: added to the pipeline template

File: components/backendappgateway/local.tf (hunk 1 above)
Comment (mokainos) [reply components/backendappgateway/local.tf:3]: There is a reference but can't find the relevant code. Bit puzzling ...
subscription_id = local.log_analytics_workspace[[for x in keys(local.log_analytics_env_mapping) : x if contains(local.log_analytics_env_mapping[x], var.env)][0]].subscription_id
}

File: azure-pipelines-frontdoor.yaml [hunk 4]
```diff
@@ -50,6 +50,8 @@ variables:
     value: $(Build.BuildNumber)
   - name: isMaster
     value: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+  - name: product
+    value 'sds-platform'
```
Comment (timja) [inline azure-pipelines-frontdoor.yaml:54]:
```suggestion
    value: 'sds-platform'
```

File: azure-pipelines.yaml [hunk 5]
```diff
@@ -68,7 +68,8 @@ variables:
     value: $(Build.BuildNumber)
   - name: isMaster
     value: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
-
+  - name: product
+    value 'sds-platform'
```
Comment (timja) [inline azure-pipelines.yaml:72]:
```suggestion
    value: 'sds-platform'
```

**Review by timja — APPROVED** [review]
